shared/devices/device-revoke/index.js

```diff
@@ -1,103 +1,195 @@
+    desktop: isMobile ? 'icon-computer-revoke-64' : 'icon-computer-revoke-48',
+    mobile: isMobile ? 'icon-phone-revoke-64' : 'icon-phone-revoke-48',
+  }[props.type]
+  return (
+    <Box style={styles.headerContainer}>
+      <Icon type={headerIcon} />
+      <Text type="BodySemibold" style={styles.headerDeviceName}>
+        {props.name}
+      </Text>
+    </Box>
+  )
+}
+
+const BodyText = (props: {name: string, currentDevice: boolean}) => (
+  <Box style={styles.bodyTextContainer}>
+    <Text type="BodySemibold" style={styles.bodyText}>
+      Are you sure you want to revoke{' '}
+      {props.currentDevice ? 'your current device' : <Text type="BodySemiboldItalic">{props.name}</Text>}
+      ?
     </Text>
   </Box>
 )
 
-const Body = ({endangeredTLFs, name, currentDevice}) => (
-  <Box>
-    <Box style={styleHeader}>
-      <Text type="BodySemibold">Are you sure you want to revoke </Text>
-      {currentDevice ? (
-        <Text type="BodySemibold">your current device</Text>
-      ) : (
-        <Text type="BodySemiboldItalic">{name}</Text>
-      )}
-      <Text type="BodySemibold">?</Text>
+class EndangeredTLFList extends React.Component<
+  {endangeredTLFs: Array<string>, waiting: boolean, style: StylesCrossPlatform},
+  {}
+> {
+  _renderTLFEntry = (index: number, tlf: string) => (
+    <Box key={index} style={styles.tlfEntry}>
+      <Text type="BodySemibold" style={{marginRight: globalMargins.tiny}}>
+        •
+      </Text>
+      <Text type="BodySemibold" selectable={true} style={{flex: 1}}>
+        {tlf}
+      </Text>
     </Box>
+  )
+  render() {
+    if (this.props.waiting) {
+      return <ProgressIndicator />
+    } else if (this.props.endangeredTLFs.length > 0) {
+      return (
+        <React.Fragment>
+          <Text type="Body" style={styles.bodyText}>
+            You may lose access to these folders forever:
+          </Text>
+          <Box style={styles.tlfListContainer}>
+            <List
+              items={this.props.endangeredTLFs}
+              renderItem={this._renderTLFEntry}
+              style={styles.tlfList}
+            />
+          </Box>
+        </React.Fragment>
+      )
+    }
+    return null
+  }
+}
 
-    {endangeredTLFs.length > 0 && (
-      <Box>
-        <Box>
-          <Text type="Body">You may lose access to these folders forever:</Text>
-        </Box>
-
-        <Box style={styleDevicesContainer}>
-          {endangeredTLFs.map(tlf => (
-            <Box key={tlf} style={styleTLF}>
-              <Text type="BodySemibold" style={{marginRight: globalMargins.tiny}}>
-                •
-              </Text>
-              <Text type="BodySemibold">{tlf}</Text>
-            </Box>
-          ))}
-        </Box>
-      </Box>
-    )}
-  </Box>
+const ActionButtons = (props: {onCancel: () => void, onSubmit: () => void, waiting: boolean}) => (
+  <Box2 direction={isMobile ? 'vertical' : 'horizontalReverse'} style={styles.actionButtonsContainer}>
```

Comment:
This is the reason why I added `horizontalReverse` and `verticalReverse` – gives us the delete button on top in mobile and to the right on desktop!